lms/djangoapps/discussion_api/permissions.py

```diff
@@ -18,6 +20,34 @@ def _is_author_or_privileged(cc_content, context):
     return context["is_requester_privileged"] or _is_author(cc_content, context)
 
 
+NON_UPDATABLE_THREAD_FIELDS = {"course_id"}
+NON_UPDATABLE_COMMENT_FIELDS = {"thread_id", "parent_id"}
+
+
+def get_initializable_thread_fields(context):
+    """
+    Return the set of fields that the requester can initialize for a thread
+    """
+    ret = get_editable_fields(
+        Thread(user_id=context["cc_requester"]["id"], type="thread"),
```

Comment:
It is a bit round about but I think the roundabout part is the get_editable_fields method itself. It requires a thread to get the correct permissions to find what fields can be initialized. Though, get_editable_fields is used by other methods as well. I'm sure there can be some more refactoring done but I think for get_initializable_thread_fields, refactoring  get_editable_fields is out of the scope of this PR.